src/components/QuickOpenModal.js

```diff
@@ -422,16 +430,21 @@ export class QuickOpenModal extends Component<Props, State> {
 /* istanbul ignore next: ignoring testing of redux connection stuff */
 function mapStateToProps(state) {
   const selectedSource = getSelectedSource(state);
+  const tabs = getTabs(state);
 
   return {
     enabled: getQuickOpenEnabled(state),
-    sources: formatSources(getDisplayedSourcesList(state), getTabs(state)),
     selectedSource,
     symbols: formatSymbols(getSymbols(state, selectedSource)),
     symbolsLoading: isSymbolsLoading(state, selectedSource),
     query: getQuickOpenQuery(state),
     searchType: getQuickOpenType(state),
-    tabs: getTabs(state)
+    tabs,
+    computeSources: (
+      tabUrls: Set<$PropertyType<Tab, "url">>
+    ): Array<Object> => {
+      return formatSources(getDisplayedSourcesList(state), tabUrls);
+    }
```

Comment:
including functions in MSPs can trigger unnecessary re-renders. 

What is it trying to do?

Reply:
This used to be computed in `mapStateToProps`, but if we kept it there we were computing it way too often (making it a bottleneck). The first approach was to declare the function at the top of the file, but then the component has a dependency on the entire state, making it hard to test (you need to make parts of the state). So I decided to keep it in `mapStateToProps` but make it be computed on demand.

You're right about the re-renderes, I think that by declaring the function once and always passing it (the same instance) via props should solve that, correct?

Reply:
> You're right about the re-renderes, I think that by declaring the function once and always passing it (the same instance) via props should solve that, correct?

 I think so. Also, you should be safe to use `getDisplayedSourcesList(state)` in the MSP because that is memoized and then call formatSources where ever you'd like...